code/modules/spells/spell_types/pointed/swap.dm

```diff
@@ -0,0 +1,82 @@
+/datum/action/cooldown/spell/pointed/swap
+	name = "Swap"
+	desc = "This spell allows you to swap locations with any living being."
+	button_icon_state = "swap"
+	ranged_mousepointer = 'icons/effects/mouse_pointers/swap_target.dmi'
+
+	school = SCHOOL_TRANSLOCATION
+	cooldown_time = 30 SECONDS
+	cooldown_reduction_per_rank = 6 SECONDS
+	cast_range = 9
+	invocation = "FRO' BRT'TRO, DA!"
+	invocation_type = INVOCATION_SHOUT
+	spell_requirements = SPELL_REQUIRES_NO_ANTIMAGIC|SPELL_REQUIRES_OFF_CENTCOM
+	active_msg = "You prepare to swap locations with a target..."
+
+	smoke_type = /datum/effect_system/fluid_spread/smoke
+	smoke_amt = 0
+	/// A variable for holding the second selected target with right click.
+	var/mob/living/second_target
+
+/datum/action/cooldown/spell/pointed/swap/is_valid_target(atom/cast_on)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(!isliving(cast_on))
+		to_chat(owner, span_warning("You can only swap locations with living beings!"))
+		return FALSE
+	return TRUE
+
+/datum/action/cooldown/spell/pointed/swap/InterceptClickOn(mob/living/caller, params, atom/click_target)
+	if(LAZYACCESS(params2list(params), RIGHT_CLICK))
+		if(!IsAvailable())
+			return FALSE
+		if(!target)
+			return FALSE
+		if(!isliving(click_target) && isturf(click_target))
+			// Find any living being in the list. We aren't picky, it's aim assist after all
+			click_target = locate(/mob/living) in click_target
+			if(!click_target)
+				to_chat(owner, span_warning("You can only select living beings as secondary target!"))
+				return FALSE
+		if(click_target == owner)
```

Comment:
would this be better as RMB while you have a target, instead?

Reply:
If someone is moving they might want to click more than once to make sure they have marked the target, that is why to unmark it you will have to click yourself.